### ISSUE TITLE
Unable to run func tasks after updating to VS Code 1.53.0

### ISSUE THREAD

**Issue by justinneff** [microsoft/vscode-azurefunctions#2654]
After installing the VS Code 1.53.0 update, I am now unable to run func related tasks. The error message received is different depending on how the task is run.

When run from the command palette, I get this message in an error toast:

```
There is no task provider registered for tasks of type "func".
```

When task used as `preLaunchTask` this message is in the Tasks Output:

```
Error: The func task detection didn't contribute a task for the following configuration:
{
    "type": "func",
    "label": "runApiFunctionsHost",
    "command": "host start",
    "problemMatcher": "$func-node-watch",
    "isBackground": true,
    "dependsOn": "api build",
    "options": {
        "cwd": "${workspaceFolder}/services/api"
    }
}
The task will be ignored.
```

I have tried uninstalling and reinstalling the Azure Functions extension, but that did not help. This configuration was working prior to the VS Code update so I am assuming that is somehow related. For reference, I have included the relevant portions of the tasks and launch configurations.

**tasks.json**

```
  {
    "type": "func",
    "command": "host start",
    "problemMatcher": "$func-node-watch",
    "isBackground": true,
    "dependsOn": "api build",
    "options": { "cwd": "${workspaceFolder}/services/api" }
  },
```

**launch.json**

```
    {
      "name": "Attach to Node Functions",
      "type": "node",
      "request": "attach",
      "port": 9229,
      "preLaunchTask": "runApiFunctionsHost",
      "cwd": "${workspaceFolder}/services/api"
    }
```

**Comment by ejizba** [comment]
Yeah looks like this is [related to](https://github.com/microsoft/vscode-azurefunctions/issues/2611) how our extension is activated. As a workaround, you can downgrade VS Code to 1.52 or just make sure you open the Azure view each time you open VS Code, which will activate our extension. This is only a problem when your project is several folders down in the workspace - projects in the root or one level down should still activate us automatically.

We'll look into the proper fix

**Comment by justinneff** [comment]
Thanks for the workaround!

**Comment by ejizba** [comment]
We released v1.2.1, which includes a fix for this. @justinneff thanks for filing!
